src/pull.go

```diff
@@ -82,6 +88,72 @@ func (g *Commands) Pull(byId bool) error {
 	return g.playPullChanges(nonConflicts, g.opts.Exports, opMap)
 }
 
+func autoRenameClashes(g *Commands, clashes []*Change) error {
+	type rename struct {
+		name string
+		change *Change
+	}
+	clashesMap := map[string][]rename{}
+
+	for _, clash := range clashes {
+		group := clashesMap[clash.Path]
+		if clash.Src == nil { // Is that even possible? In what situation?
+			continue
+		}
+		clashesMap[clash.Path] = append(group, rename{"", clash})
+	}
+
+	for path, group := range clashesMap {
+		ext := filepath.Ext(path)
+		name := strings.TrimSuffix(path, ext)
+		n := 0
+
+		for i, r := range group[1:] {  // we can leave first item with original name
```

Comment:
Using i, r := range group[1:] means we have to remember that we are ignoring the first element always, doing this more than once can cause bugs if we forget. after that iteration, add elements retrieved into a fresh list.